src/uwtools/drivers/forecast.py

```diff
@@ -126,10 +131,64 @@ def resources(self) -> None:
         ???
         """
 
-    def run(self):
+    def run(
+        self,
+        config_file: str,
+        forecast_app: str,
+        forecast_model: str,
+        dry_run: bool = False,
+        log_file: Optional[str] = None,
+        machine: Optional[str] = None,
+        quiet: bool = False,
+        verbose: bool = False,
+    ) -> None:  # pragma: no cover
         """
         Runs FV3.
         """
+        # set up logging
+        name = f"{inspect.stack()[0][3]}"
+        log = cli_helpers.setup_logging(
+            log_file=log_file or "/dev/stdout",
+            log_name=name,
+            quiet=quiet,
+            verbose=verbose,
+        )
+        # read in the config file
+        # self.create_model_config(config_file=config_file)
+        experiment_config = config.YAMLConfig(config_file)
+        # prepare directories
+        run_directory = experiment_config["parameters"]["run_directory"]
+        self.create_directory_structure(run_directory)
+
+        static_files = experiment_config["locations"]["static"]
+        self.stage_files(run_directory, static_files, link_files=False)
+        cycledep_files = experiment_config["locations"]["cycledep"]
+        self.stage_files(run_directory, cycledep_files, link_files=True)
+
+        experiment_resources = {
+            key: experiment_config["resources"][key] for key in experiment_config["resources"]
+        }
+        batch_script = self.batch_script(experiment_resources)
+        args = str(experiment_config["run_cmd"]["kwargs"])
+        run_command = self.run_cmd(
+            args,
+            run_cmd=experiment_config["run_cmd"]["run_cmd"],
+            exec_name=experiment_config["run_cmd"]["exec_name"],
+        )
+
+        if dry_run:
+            # Apply switch to allow user to view the run command of config
+            # This will not run the job.
+            log.info(f"Configuration: {forecast_app} {forecast_model} {machine}")
+            log.info(f"Run command: {run_command} {batch_script}")
+            return
+        # run the job
+        subprocess.run(
```

Comment:
I think right around here is where the script should make a decision about whether to run the mpi command or submit a batch script that runs the mpi command.

Reply:
I agree, but what would that decision be based on? Should that be handled at the cli level, e.g. `uw forecast run ...` vs `uw forecast submit ...`? Or do we already know the user's intent some other way?

Reply:
My preference is at the cli level, though I can set some structure here